wqflask/wqflask/correlation/show_corr_results.py

```diff
@@ -86,139 +273,18 @@ def correlation_json_for_table(start_vars, correlation_data, this_trait, this_da
-            results_dict['lrs_location'] = target_trait['lrs_location']
-            if bool(target_trait['additive']):
-                results_dict['additive'] = f"{float(target_trait['additive']):.3f}"
-            results_dict['lit_corr'] = "--"
-            results_dict['tissue_corr'] = "--"
-            results_dict['tissue_pvalue'] = "--"
-            if this_dataset['type'] == "ProbeSet":
-                if 'lit_corr' in trait:
-                    results_dict['lit_corr'] = f"{float(trait['lit_corr']):.3f}"
-                if 'tissue_corr' in trait:
-                    results_dict['tissue_corr'] = f"{float(trait['tissue_corr']):.3f}"
-                    results_dict['tissue_pvalue'] = f"{float(trait['tissue_p_val']):.3e}"
-        elif target_dataset['type'] == "Publish":
-            results_dict['abbreviation_display'] = "N/A"
-            results_dict['description'] = "N/A"
-            results_dict['mean'] = "N/A"
-            results_dict['authors_display'] = "N/A"
-            results_dict['additive'] = "N/A"
-            results_dict['pubmed_link'] = "N/A"
-            results_dict['pubmed_text'] = target_trait["pubmed_text"]
-
-            if bool(target_trait['abbreviation']):
-                results_dict['abbreviation_display'] = target_trait['abbreviation']
-            if bool(target_trait['description']):
-                results_dict['description'] = target_trait['description']
-            if bool(target_trait['mean']):
-                results_dict['mean'] = f"{float(target_trait['mean']):.3f}"
-            if bool(target_trait['authors']):
-                authors_list = target_trait['authors'].split(',')
-                if len(authors_list) > 6:
-                    results_dict['authors_display'] = ", ".join(
-                        authors_list[:6]) + ", et al."
-                else:
-                    results_dict['authors_display'] = target_trait['authors']
-            if 'pubmed_id' in target_trait:
-                results_dict['pubmed_link'] = target_trait['pubmed_link']
-                results_dict['pubmed_text'] = target_trait['pubmed_text']
-            try:
-                results_dict['lod_score'] = f"{float(target_trait['lrs_score']) / 4.61:.1f}"
-            except:
-                results_dict['lod_score'] = "N/A"
-            results_dict['lrs_location'] = target_trait['lrs_location']
-            if bool(target_trait['additive']):
-                results_dict['additive'] = f"{float(target_trait['additive']):.3f}"
-        else:
-            results_dict['location'] = target_trait['location']
 
-        results_list.append(results_dict)
+    dataset_metadata = generate_table_metadata({name for trait in corr_results
+                                                for (name, _val) in trait.items()},
+                                               correlation_data["traits_metadata"],
+                                               target_dataset_ob)
```

Comment:
This nested "for" loop would be difficult to maintain.  I'd advise using a more simpler, albeit verbose, loop instead.